### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by Gambit142**
In this branch, we did the following:

- Created a class `MyList` that has an instance variable `@list`.
- In the `MyList`, we implemented a method `#each` that yields successive members of `@list` and uses the MyEnumerable module.
- Create a module `MyEnumerable` that implements the following methods:
  - #all?
  - #any?
  - #filter
  - #max
  - #min
  - #sort
- Each class and module had a separate `.rb` file.
- Verified our solutions using these checklists;
```
# Create our list
irb> list = MyList.new(1, 2, 3, 4)
=> #<MyList: @list=[1, 2, 3, 4]>

# Test #all?
irb> list.all? {|e| e < 5}
=> true
irb> list.all? {|e| e > 5}
=> false

# Test #any?
irb> list.any? {|e| e == 2}
=> true
irb> list.any? {|e| e == 5}
=> false

# Test #filter
irb> list.filter {|e| e.even?}
=> [2, 4]
```

### CODE REVIEW

File: mylist.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative 'enumerable'
+class MyList
+  include MyEnumerable
+  def initialize(*args)
+    @list = args
+  end
```
Comment (Emmyn5600) [inline mylist.rb:6]: - [ ] Optional suggestions

- `#initialize` method accepts multiple arguments. so I would recommend doing this

````
def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```
`

File: mylist.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  def each
+    i = 0
+    while @list[i]
+      yield @list[i]
+      i += 1
+    end
+  end
```
Comment (Emmyn5600) [inline mylist.rb:14]: - [ ] Optional suggestions

You should return a normal Enumerator using to_enum, so I would recommend adding the below line before I = 0 declaration

- `return to_enum(:each) unless block_given?` and return @list after end of while.

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :+1: 

Hello, the team @Gambit142 and @Busiwa24    👋🏾 👋🏾 congratulations on implementing all the project requirements.

## Optional suggestions

- _Check the comment under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

File: mylist.rb (hunk 1 above)
Comment (MrBrN197) [reply mylist.rb:6]: `*args` will accept all the arguments; from my understanding therefore `args` itself will be an array. I believe What they have already is more optimal.

Comment (Gambit142) [reply mylist.rb:6]: Thanks for the feedback

File: mylist.rb (hunk 2 above)
Comment (Gambit142) [reply mylist.rb:14]: Thanks for the feedback

**Pull request merged**